src/libraries/System.Collections.Concurrent/src/System/Collections/Concurrent/BlockingCollection.cs

```diff
@@ -1709,8 +1709,7 @@ private static void ValidateCollectionsArray(BlockingCollection<T>[] collections
                 throw new ArgumentException(
                     SR.BlockingCollection_ValidateCollectionsArray_ZeroSize, nameof(collections));
             }
-            else if ((!IsSTAThread && collections.Length > 63) || (IsSTAThread && collections.Length > 62))
-            //The number of WaitHandles must be <= 64 for MTA, and <=63 for STA, and we reserve one for CancellationToken
+            else if (collections.Length > 63) //The number of WaitHandles for WaitHandle.WaitAny must be <= 64, and we reserve one for CancellationToken
```

Comment:
Should this actually be > 62 in case we're on an STA and have a cancellation token?